### ISSUE TITLE
How to distribute weights to different GPUs?

### ISSUE THREAD

**Issue by w32zhong** [OpenBMB/BMTrain#124]
I have tried to load the checkpoints through:
```sh
    local_rank = int(os.environ["LOCAL_RANK"])

    bmtrain.init_distributed(seed=seed)

    rank_print('loading tokenizer...')
    tokenizer = LlamaTokenizer.from_pretrained(token_path)

    rank_print('allocating model...')
    config = LlamaConfig.from_pretrained(model_path)
    model = LlamaForCausalLM(config) # allocate space
    model = bmtrain.BMTrainModelWrapper(model)
    rank_print('allocated.')

    rank_print('loading checkpoints from rank0 ...')
    ckpt = os.path.join(model_path, "state_dict.pt")
    load_res = bmtrain.load(model, ckpt, strict=True)
    rank_print('Loaded:', load_res)

```
However, I find the `bmtrain.load` will load `ckpt` to each process. Am I supposed to use `local_rank` to `bmtrain.load` shared checkpoints?

**Comment by Achazwl** [comment]
`bmt.load` distribute weights to each rank, each 1/n of the ckpt.
Load ckpt to each process is a bug of bmtrain since torch 1.12.0. We are now fixing it.
Using torch<=1.11 can avoid this bug.

**Comment by Achazwl** [comment]
Fixed in PR #115.
